stripe/src/main/java/com/stripe/android/StripeKtx.kt

```diff
@@ -0,0 +1,577 @@
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [SetupIntent] object.
+ *
+ * See [Confirm a SetupIntent](https://stripe.com/docs/api/setup_intents/confirm).
+ * `POST /v1/setup_intents/:id/confirm`
+ *
+ * @param confirmSetupIntentParams a set of params with which to confirm the Setup Intent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [SetupIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmSetupIntent(
+    confirmSetupIntentParams: ConfirmSetupIntentParams,
+    idempotencyKey: String? = null
+): SetupIntent = runApiRequest {
+    stripeRepository.confirmSetupIntent(
+        confirmSetupIntentParams,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId,
+            idempotencyKey = idempotencyKey
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [PaymentIntent] object.
+ *
+ * See [Confirm a PaymentIntent](https://stripe.com/docs/api/payment_intents/confirm).
+ * `POST /v1/payment_intents/:id/confirm`
+ *
+ * @param confirmPaymentIntentParams a set of params with which to confirm the PaymentIntent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [PaymentIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmPaymentIntent(
+    confirmPaymentIntentParams: ConfirmPaymentIntentParams,
+    idempotencyKey: String? = null
+): PaymentIntent = runApiRequest {
+    stripeRepository.confirmPaymentIntent(
+        confirmPaymentIntentParams,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId,
+            idempotencyKey = idempotencyKey
+        )
+    )
+}
+
+/**
+ * Consume the empty result from Stripe's internal Json Parser, throw [InvalidRequestException] for public API.
+ */
+private suspend inline fun <reified T : Any> runApiRequest(
+    crossinline block: suspend () -> T?
```

Comment:
Do we need `crossinline`?

Reply:
No, actually even `suspend` is not needed, cleaned them up.